code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm

```diff
@@ -0,0 +1,448 @@
+			var/datum/component/oldmap = GetComponent(/datum/component/multitile)
+			if(oldmap)
+				qdel(oldmap)
+			AddComponent(/datum/component/multitile, list(
+				list(1,				1, 0),
+				list(1,				1, 1),
+				list(MACH_CENTER, 	1, 0),
+			))
+
+/obj/machinery/power/transmission_laser/Destroy()
+	. = ..()
+	if(length(laser_effects))
+		destroy_lasers()
+	blocked_objects = null
+
+/obj/machinery/power/transmission_laser/proc/get_back_turf()
+	//this is weird as i believe byond sets the bottom left corner as the source corner like
+	// x-x-x
+	// x-x-x
+	// o-x-x
+	//which would mean finding the true back turf would require centering than taking a step in the inverse direction
+	var/turf/center = locate(x + 1, y + 1, z)
+	if(!center)///what
+		return
+	var/inverse_direction = turn(dir, 180)
+	return get_step(center, inverse_direction)
+
+/obj/machinery/power/transmission_laser/proc/get_front_turf()
+	//this is weird as i believe byond sets the bottom left corner as the source corner like
+	// x-x-x
+	// x-x-x
+	// o-x-x
+	//which would mean finding the true front turf would require centering than taking a step in the primary direction
+	var/turf/center = locate(x + 1 + round(pixel_x / 32), y + 1 + round(pixel_y / 32), z)
+	if(!center)///what
+		return
+	return get_step(center, dir)
+
+/obj/machinery/power/transmission_laser/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Laser currently has [unsent_earnings] unsent credits.<span/>"
+	. += "<span class='notice'>Laser has generated [total_earnings] credits.<span/>"
+	. += "<span class='notice'>Laser has sold [total_energy] Joules<span/>"
+///appearance changes are here
+
+/obj/machinery/power/transmission_laser/update_overlays()
+	. = ..()
+	if((stat & BROKEN) || !charge)
+		. += "unpowered"
+		return
+	if(input_available > 0)
+		. += "green_light"
+		. += emissive_appearance(icon, "green_light", src)
+	if(turned_on)
+		. += "red_light"
+		. += emissive_appearance(icon, "red_light", src)
+		if(firing)
+			. +="firing"
+			. += emissive_appearance(icon, "firing", src)
+
+	var/charge_level = return_charge()
+	if(charge_level == 6)
+		. += "charge_full"
+		. += emissive_appearance(icon, "charge_full", src)
+	else if(charge_level > 0)
+		. += "charge_[charge_level]"
+		. += emissive_appearance(icon, "charge_[charge_level]", src)
+
+///returns the charge level from [0 to 6]
+/obj/machinery/power/transmission_laser/proc/return_charge()
+	if(!output_level)
+		return 0
+	return min(round((charge / abs(output_level)) * 6), 6)
+
+/obj/machinery/power/transmission_laser/proc/send_ptl_announcement()
+	/// The message we send
+	var/message
+	var/flavor_text
+	if(announcement_threshold == 1 MJ)
+		message = "PTL account successfully made"
+		flavor_text = "From now on, you will receive regular updates on the power exported via the onboard PTL. Good luck [station_name()]!"
+		announcement_threshold = 100 MJ
+
+	message = "New milestone reached!\n[DisplayJoules(announcement_threshold)]\n[flavor_text]"
```

Comment:
Would be curious to see how frequently we see these announcements pop up

Reply:
With 200MW it would be 5 seconds then a little over 4 minutes then a little under 4 hours. I need to set the multiplier much lower honestly.

Reply:
I changed the threshold to increase more reasonably and also capped the rate in which it can increase.